pandas/core/sorting.py

```diff
@@ -374,7 +374,16 @@ def nargsort(
     if not ascending:
         non_nans = non_nans[::-1]
         non_nan_idx = non_nan_idx[::-1]
-    indexer = non_nan_idx[non_nans.argsort(kind=kind)]
+
+    # GH 35922. Move support for object sort here from Series.sort_values
+    try:
+        # if kind==mergesort, it can fail for object dtype
```

Comment:
why can this fail?

Reply:
@jreback I moved this code from `Series.sort_values`, fearing that I don't understand why this is necessary, but it doesn't seem to be. I've removed the try/except, tested sorting `Series` of `object` with different `kind` settings, and ran the test suite, and nothing is going wrong. I think we can safely drop it.